### PR TITLE
Add Attributes v2 API to the NG Secret Provider

### LINKED ISSUE #84: Implement Attributes v2 common interface for Secret Provider implementation in connector
### Description

Implement the Attributes v2 common provider interface (`com.otilm.api.interfaces.connector.common.v2.AttributesController`) in the **Secret Provider (v2 / NG)** connector hosted by common-credential-provider, so its API exposes the standard NG attribute surface:

- `GET /v2/attributes` — attribute-definition registry (`listDefinitions`)
- `GET /v2/attributes/{uuid}` — single definition (`getDefinition`), `ATTRIBUTE_DEFINITION_NOT_FOUND` when unknown
- `POST /v2/attributes/callback` — stateless callback envelope

common-credential-provider runs two connectors in one service: a Credential Provider (v1, legacy common) and a Secret Provider (v2, NG framework). This issue targets **only the NG Secret Provider**; the v1 Credential Provider path is unchanged.

Connector-side counterpart to interface-documentation#211 (PR OmniTrustILM/interface-documentation#216), which added the controller to the `doc-openapi-connector-secret-provider` OpenAPI group. Part of the Attributes v2 provider interface API epic (ilm#251).

### User Story

As a common-credential-provider maintainer, I want the NG Secret Provider to implement the common Attributes v2 interface so Core can dispatch stateless secret-provider callbacks and resolve attribute definitions on demand, while the v1 Credential Provider connector remains untouched.

### Use Case

Core issues a stateless secret-provider vault-profile callback to the NG connector; it resolves dynamic attribute content and returns it via the callback envelope without per-connector state. The coexisting v1 Credential Provider connector continues to serve its `/v1` surface unchanged.

### Acceptance Criteria

- [ ] NG Secret Provider connector implements `connector.common.v2.AttributesController`.
- [ ] `GET /v2/attributes` returns the connector's attribute-definition registry.
- [ ] `GET /v2/attributes/{uuid}` returns a single definition, or `ATTRIBUTE_DEFINITION_NOT_FOUND` when absent.
- [ ] `POST /v2/attributes/callback` handles the stateless callback request/response envelope.
- [ ] NG connector OpenAPI matches the `doc-openapi-connector-secret-provider` contract in interface-documentation.
- [ ] v1 Credential Provider connector behaviour unchanged.

### PR DESCRIPTION
**Pull request by lubomirw**
Implements the common Attributes v2 provider interface (`com.otilm.api.interfaces.connector.common.v2.AttributesController`, `/v2/attributes`) for the NG Secret Provider. The v1 Credential Provider is unchanged.

Closes #84.

## What changed
- **`AttributesControllerImpl`** (`/v2/attributes`, bean `attributesControllerV2`) implementing the common interface — `listDefinitions`, `getDefinition`, `callback`.
- **`SecretProviderAttributesService`** — the connector-global attribute-definition registry. Aggregates the connector's NG definition shapes (today just `data_namespace`) and self-validates UUID uniqueness at startup.
- **`/v2/info`** now advertises the `ATTRIBUTES` interface at version `v2`.
- **Error mapping** — `ATTRIBUTE_DEFINITION_NOT_FOUND` (404) and, on the callback surface, `VALIDATION_FAILED` (422) via `ProblemDetailsHandlingAdvice`.

## Design notes
- `data_namespace` declares **no callback**: one definition shape, identical on every endpoint (definition-identity). The registry serves the **content-free shape** (no DB read); the existing `VaultController.listVaultAttributes()` form path still returns the shape plus the live namespace options.
- The callback endpoint is **inert** (this connector declares no `dependsOn` callbacks): an unknown UUID returns 404 `ATTRIBUTE_DEFINITION_NOT_FOUND`; a known-but-non-dispatchable UUID returns 422 `VALIDATION_FAILED` (non-retryable, since a registry refresh would never make it dispatchable).

## Acceptance criteria
- [x] NG Secret Provider implements `connector.common.v2.AttributesController`.
- [x] `GET /v2/attributes` returns the attribute-definition registry.
- [x] `GET /v2/attributes/{uuid}` returns a single definition, or `ATTRIBUTE_DEFINITION_NOT_FOUND` when absent.
- [x] `POST /v2/attributes/callback` handles the stateless callback request/response envelope.
- [x] NG connector OpenAPI matches the `doc-openapi-connector-secret-provider` contract — satisfied by implementing the annotated interface (the spec is generated centrally in interface-documentation, where #216 added the controller to that group).
- [x] v1 Credential Provider behaviour unchanged.

## Tests
Full suite green (114/114). Unit coverage for the registry (list / get / filter / duplicate-UUID self-validation), the inert-callback 404-vs-422 split, the content-free / no-DB contract, controller delegation, the error-code mappings, and info advertisement; the context test confirms the controller is wired and advised.

## Follow-ups (not in this PR)
- End-to-end MockMvc web-slice test — the repo has no MockMvc harness today (error mapping is unit-tested, wiring is context-tested).
- The Connector-NG "Attributes" concept docs still describe the old per-operation pattern (documentation repo — org-wide, tied to the interfaces Attributes v2 rollout).

### CODE REVIEW

File: src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    @PostConstruct
+    void validateRegistry() {
+        Set<String> seen = new HashSet<>();
+        for (BaseAttribute definition : allDefinitions()) {
+            if (!seen.add(definition.getUuid())) {
+                throw new IllegalStateException(
+                        "Duplicate attribute UUID in connector registry: " + definition.getUuid());
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (lubomirw) [reply src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java:50]: Fixed in c3cd853 — registry now matches attribute UUIDs by value (canonical-form-independent); javadoc corrected.

File: src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    @Override
+    public AttributeDefinitionsDto listDefinitions(List<UUID> uuids) {
+        List<BaseAttribute> definitions = allDefinitions();
+        if (uuids != null && !uuids.isEmpty()) {
+            Set<String> wanted = new HashSet<>();
+            uuids.forEach(uuid -> wanted.add(uuid.toString()));
+            definitions = definitions.stream()
+                    .filter(definition -> wanted.contains(definition.getUuid()))
+                    .toList();
+        }
```
[the remark this answers is not in the document]
Comment (lubomirw) [reply src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java:71]: Fixed in c3cd853 — registry now matches attribute UUIDs by value (canonical-form-independent); javadoc corrected.

File: src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    @Override
+    public BaseAttribute getDefinition(UUID uuid) {
+        String key = uuid.toString();
+        return allDefinitions().stream()
+                .filter(definition -> key.equals(definition.getUuid()))
+                .findFirst()
+                .orElseThrow(() -> new AttributeDefinitionNotFoundException(uuid));
+    }
```
[the remark this answers is not in the document]
Comment (lubomirw) [reply src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java:84]: Fixed in c3cd853 — registry now matches attribute UUIDs by value (canonical-form-independent); javadoc corrected.

File: src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                .anyMatch(definition -> key.equals(definition.getUuid()));
+        if (!known) {
+            // Unknown to this connector — 404; Core refreshing its registry is a sensible reaction.
+            throw new AttributeDefinitionNotFoundException(uuid);
+        }
+        // Known, but this connector declares no NG callback (dependsOn), so nothing is dispatchable;
+        // a registry refresh would not change that, so reject as non-retryable rather than not-found.
+        // In normal operation Core never calls this — it only dispatches attributes advertising dependsOn.
+        throw new AttributeCallbackNotSupportedException(uuid);
+    }
```
[the remark this answers is not in the document]
Comment (lubomirw) [reply src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java:99]: Fixed in c3cd853 — registry now matches attribute UUIDs by value (canonical-form-independent); javadoc corrected.

File: src/main/java/com/otilm/common/credential/provider/secret/api/v2/AttributeDefinitionNotFoundException.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package com.otilm.common.credential.provider.secret.api.v2;
+
+import java.util.UUID;
+
+/**
+ * Thrown when a requested attribute UUID is not a resolvable definition on this connector —
+ * unknown to the registry, or (on the callback surface) present but not callback-dispatchable.
+ * Unchecked because the {@code AttributesController} interface methods declare no checked exceptions.
+ */
```
[the remark this answers is not in the document]
Comment (lubomirw) [reply src/main/java/com/otilm/common/credential/provider/secret/api/v2/AttributeDefinitionNotFoundException.java:10]: Fixed in c3cd853 — registry now matches attribute UUIDs by value (canonical-form-independent); javadoc corrected.

File: src/test/java/com/otilm/common/credential/provider/secret/service/VaultAttributeServiceTest.java [hunk 6]
```diff
@@ -43,4 +43,23 @@ void listVaultAttributes_returnsExtensibleNamespaceListOfExistingNamespaces() {
...
         Assertions.assertEquals(List.of(), vaultAttributeService.listVaultProfileAttributes());
     }
+
+    @Test
+    void namespaceAttributeDeclaresNoCallback() {
+        DataAttributeV3 namespace =
+                (DataAttributeV3) vaultAttributeService.listVaultAttributes().get(0);
+        Assertions.assertNull(namespace.getAttributeCallback(),
+                "data_namespace must declare no callback — options are baked extensible-list content");
+    }
```
[the remark this answers is not in the document]
Comment (lubomirw) [reply src/test/java/com/otilm/common/credential/provider/secret/service/VaultAttributeServiceTest.java:54]: Not an issue: Mockito returns an empty list (not null) for unstubbed collection-returning methods, so listVaultAttributes()'s findDistinctNamespaces().stream() is safe — no NPE. The suite is green (115/115). This test asserts only the absence of a callback, so it intentionally does not stub content.

File: src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+                .anyMatch(definition -> uuid.equals(definitionUuid(definition)));
+        if (!known) {
+            // Unknown to this connector — 404; Core refreshing its registry is a sensible reaction.
+            throw new AttributeDefinitionNotFoundException(uuid);
+        }
+        // The attribute is known but declares no NG callback, so nothing can be resolved. A registry
+        // refresh would not change that, so this is a non-retryable rejection rather than a not-found.
+        // Core does not reach here in normal operation — it dispatches only callback-enabled attributes.
+        throw new AttributeCallbackNotSupportedException(uuid);
+    }
```
[the remark this answers is not in the document]
Comment (lubomirw) [reply src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java:99]: callback() is reached only via @Valid @RequestBody AttributeCallbackRequestDto, whose attributeUuid is @NotNull. Spring rejects a null/absent attributeUuid (or an unreadable body) as 422 VALIDATION_FAILED before the method body runs, so the described 500 path does not occur on the wire. A service-layer null guard would only duplicate the bean-validation contract; leaving as-is.

File: src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    public SecretProviderAttributesServiceImpl(BuildProperties buildProperties,
+                                               VaultAttributeService vaultAttributeService) {
+        this.buildProperties = buildProperties;
+        this.vaultAttributeService = vaultAttributeService;
+    }
+
+    /**
+     * Startup self-validation: fail fast if the connector-global registry contains a malformed or
+     * duplicate UUID. This connector declares no callbacks, so there is no callback-dispatchability
+     * check here yet — add one alongside the first attribute that declares a callback.
```
Comment (ivosh) [inline src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java]: **. Nitpick** | Documentation

In SecretProviderAttributesServiceImpl.java:35 the validateRegistry() javadoc ends with an imperative future-work note: "...so there is no callback-dispatchability check here yet — add one alongside the first attribute that declares a callback." The first two sentences are a genuine WHY (explaining why the dispatchability check is absent today) and are fine to keep. The trailing "add one alongside..." reads as a soft TODO baked into prose, which tends to go stale and won't surface anywhere a TODO scan would find it. Two readings: (a) intentional design note documenting current state — leave as is; (b) really a reminder for future work — in that case consider dropping the imperative clause or moving it to a tracked ticket, keeping the javadoc to the WHY. Non-blocking either way.

File: src/test/java/com/otilm/common/credential/provider/secret/service/VaultAttributeServiceTest.java [hunk 9]
```diff
@@ -43,4 +43,23 @@ void listVaultAttributes_returnsExtensibleNamespaceListOfExistingNamespaces() {
...
+
+    @Test
+    void listVaultAttributeDefinitions_hasNoResolvedContentAndDoesNotQueryNamespaces() {
+        DataAttributeV3 namespace =
+                (DataAttributeV3) vaultAttributeService.listVaultAttributeDefinitions().get(0);
+
+        Assertions.assertEquals(VaultAttributeServiceImpl.ATTRIBUTE_NAMESPACE, namespace.getName());
+        Assertions.assertTrue(namespace.getProperties().isExtensibleList());
+        Assertions.assertNull(namespace.getContent(), "registry definition carries no resolved option content");
+        org.mockito.Mockito.verifyNoInteractions(secretRepository);
```
Comment (ivosh) [inline src/test/java/com/otilm/common/credential/provider/secret/service/VaultAttributeServiceTest.java]: **~ Minor** | Maintainability

At `VaultAttributeServiceTest.java:63` the new test calls `org.mockito.Mockito.verifyNoInteractions(secretRepository);` with the class fully qualified inline. This departs from Google Java Style 3.3.1 (import the type; do not use FQNs in the body) and from the file's own established convention: lines 14-15 already static-import the other Mockito entry points (`import static org.mockito.Mockito.mock;` / `when;`). Add `import static org.mockito.Mockito.verifyNoInteractions;` and call `verifyNoInteractions(secretRepository);` to match. Minor, cosmetic — compiles and runs correctly.

File: src/main/java/com/otilm/common/credential/provider/secret/api/v2/AttributeDefinitionNotFoundException.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Thrown when a requested attribute UUID is unknown to this connector's registry (no matching
+ * definition). A known attribute that merely declares no callback is a different case — see
+ * {@link AttributeCallbackNotSupportedException}. Unchecked because the {@code AttributesController}
+ * interface methods declare no checked exceptions.
+ */
+public class AttributeDefinitionNotFoundException extends RuntimeException {
+
+    private final transient UUID uuid;
```
Comment (ivosh) [inline src/main/java/com/otilm/common/credential/provider/secret/api/v2/AttributeDefinitionNotFoundException.java]: **. Nitpick** | Correctness

Both new exceptions declare `private final transient UUID uuid;` (`AttributeDefinitionNotFoundException.java:13` and `AttributeCallbackNotSupportedException.java:14`). `RuntimeException`/`Throwable` is `Serializable`, and `java.util.UUID` is itself `Serializable`, so the field does not need `transient` to keep the exception serializable — no SonarQube S1948-style warning applies. Marking it `transient` means the reverse: if either exception is ever serialized/deserialized, `uuid` is excluded and `getUuid()` returns `null` (there is no `readObject`/`readResolve` to restore it), and neither class declares a `serialVersionUID`. Two readings: (a) intentional — these exceptions are only ever handled in-process by `ProblemDetailsHandlingAdvice` and never cross a wire, so the loss is harmless and `transient` is just noise; action: drop `transient` since it protects against nothing and only introduces a latent null. (b) a copy of a reflex used for genuinely non-serializable fields; action: same — remove `transient`. Either way the keyword is unnecessary here and mildly misleading. Nitpick, no runtime impact on the current in-process flow.

File: src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+        }
+    }
+
+    /**
+     * Every NG attribute definition this connector exposes. Only vault-instance attributes are
+     * non-empty today; the other NG sources (vault-profile, per-secret-type, rotate) return empty,
+     * so aggregating {@code listVaultAttributeDefinitions()} is complete. New attributes appear here
+     * once their owner returns them.
+     */
+    private List<BaseAttribute> allDefinitions() {
```
Comment (ivosh) [inline src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java:59]: **. Nitpick** | Testing

— `src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java:58` (`allDefinitions()`): the registry aggregates only `vaultAttributeService.listVaultAttributeDefinitions()`. This is correct today only because the other three NG attribute sources happen to return empty, but nothing enforces that. `validateRegistry()` would not catch a future maintainer making `getSecretAttributes()` non-empty without wiring it in here — the new attributes would silently 404 from `getDefinition` and be missing from Core's registry sync. Concrete fix: either aggregate all sources here (they're cheap empty lists), or add a unit test asserting the other three sources are empty so the assumption fails loudly if it ever changes.

**Review by ivosh — DISMISSED** [review]

File: src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java (hunk 8 above)
Comment (lubomirw) [reply src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java]: Fixed in 48c6fbd — dropped the trailing future-work clause; javadoc now states only the WHY.

File: src/test/java/com/otilm/common/credential/provider/secret/service/VaultAttributeServiceTest.java (hunk 9 above)
Comment (lubomirw) [reply src/test/java/com/otilm/common/credential/provider/secret/service/VaultAttributeServiceTest.java]: Fixed in 48c6fbd — added `import static org.mockito.Mockito.verifyNoInteractions;` and use the short form.

File: src/main/java/com/otilm/common/credential/provider/secret/api/v2/AttributeDefinitionNotFoundException.java (hunk 10 above)
Comment (lubomirw) [reply src/main/java/com/otilm/common/credential/provider/secret/api/v2/AttributeDefinitionNotFoundException.java]: Fixed in 48c6fbd — removed `transient` from the uuid field in both exceptions (UUID is Serializable; the keyword only introduced a latent null after deserialization).

File: src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java (hunk 11 above)
Comment (lubomirw) [reply src/main/java/com/otilm/common/credential/provider/secret/service/impl/SecretProviderAttributesServiceImpl.java:59]: Fixed in 48c6fbd — corrected the misleading javadoc (it claimed new attributes appear automatically). The three currently-empty sources are already pinned empty by tests (testGetRotateAttributes/testGetSecretAttributes/listVaultProfileAttributes_isEmpty), so the assumption fails loudly if violated; the javadoc now says a newly-non-empty source must be wired in here.

**Review by ivosh — APPROVED** [review]
